modules/finality-verifier/src/lib.rs

```diff
@@ -142,13 +154,86 @@ pub mod pallet {
 				<Error<T>>::InvalidAncestryProof
 			);
 
-			let _ = T::HeaderChain::append_header(finality_target)?;
+			let _ = T::HeaderChain::append_header(finality_target.clone())?;
 			frame_support::debug::info!("Succesfully imported finalized header with hash {:?}!", hash);
```

Comment:
This is only partially correct now, since `import_header` might fail. We may also end up with header being appended to `HeaderChain`, but not imported locally.